### PR TITLE
The ProductManager now communicates with other modules

### PR DESCRIPTION
**Pull request by marc7636**
This also includes a bug fix for how our products handle null values

### CODE REVIEW

**Comment by marc7636** [comment]
Repo wide error is now gone and the Test Suite has been fixed by using a newer version of JUnit

**Comment by marc7636** [comment]
We have been given the go-ahead with updating the version of JUnit to 5.8.2 during the Project Group Meeting (?)

File: src/main/java/module-info.java [hunk 1]
```diff
@@ -17,4 +17,4 @@
     requires org.jetbrains.annotations;
     opens dk.sdu.se_f22 to javafx.fxml;
     exports dk.sdu.se_f22;
-}
+}
```
Comment (MagnusHJensen) [inline src/main/java/module-info.java:20]: Unecessary change.

File: src/test/java/dk/sdu/se_f22/productmodule/management/TestSuite.java [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+
 import org.junit.platform.suite.api.SelectClasses;
 import org.junit.platform.suite.api.SelectPackages;
 import org.junit.platform.suite.api.Suite;
 
 @Suite
-//@SelectClasses({ProductManagerTest.class, ProductAttributeTest.class, JSONReaderTest.class, ProductTest.class})
-@SelectPackages({"dk.sdu.se_f22.productmodule.management"})
+@SelectClasses({ProductManagerTest.class, ProductAttributeTest.class, JSONReaderTest.class, ProductTest.class})
+//@SelectPackages({"dk.sdu.se_f22.productmodule.management"})
```
Comment (MagnusHJensen) [inline src/test/java/dk/sdu/se_f22/productmodule/management/TestSuite.java:9]: If commented out shouldn't it be removed?

File: src/test/resources/dk/sdu/se_f22/productmodule/management/products.json [hunk 3]
```diff
@@ -1,713 +1,713 @@
 [
```
Comment (MagnusHJensen) [inline src/test/resources/dk/sdu/se_f22/productmodule/management/products.json:1]: Improper formatting on this file, no reason to change it imo.

File: src/main/java/dk/sdu/se_f22/productmodule/management/ProductJSONReader.java [hunk 4]
```diff
@@ -227,7 +226,6 @@ private int findLastOccurence(String whatToFind, String line){
             index--;
         }
```
Comment (OthelloEngineer) [inline src/main/java/dk/sdu/se_f22/productmodule/management/ProductJSONReader.java:227]: Kunne man ikke bruge "index" istedet for "i" her, da den dekrementeres sammen med i

**Review by OthelloEngineer — APPROVED** [review]
En enkelt kommentar, men ellers ser det fint ud

File: src/test/java/dk/sdu/se_f22/productmodule/management/TestSuite.java (hunk 2 above)
Comment (marc7636) [reply src/test/java/dk/sdu/se_f22/productmodule/management/TestSuite.java:9]: we have been using both for testing

**Pull request merged**